packages/compass/src/main/application.ts

```diff
@@ -54,6 +72,27 @@ class CompassApplication {
     this.setupLifecycleListeners();
     this.setupApplicationMenu();
     this.setupWindowManager();
+
+    const {
+      protectConnectionStrings,
+      readOnly,
+      file,
+      positionalArguments,
+      // TODO: COMPASS-6063
+      // maxTimeMS,
+    } = preferences.getPreferences();
+
+    debug('application launched');
+    track('Application Launched', {
+      context: getContext(mode),
+      launch_connection: getLaunchConnectionSource(file, positionalArguments),
+      protected: protectConnectionStrings,
+      readOnly,
+      // TODO: replace with maxTimeMS from preferences COMPASS-6063.
+      maxTimeMS: DEFAULT_MAX_TIME_MS,
```

Comment:
The default for the preference value is going to be `undefined`, though, since this isn’t the maxTimeMS value that’s used in queries, it’s the _upper limit_ for that value (even though it’s also called `maxTimeMS`).

Reply:
So maxTimeMS in settings is the default value in input and DEFAULT_MAX_TIME_MS in query / aggregations / schema is the upper limit? do you think maybe it would be nice to rename this DEFAULT_MAX_TIME_MS for readability?

Reply:
Err, no, it’s the other way around – maxTimeMS from settings is the upper limit, because the goal is that admins are supposed to be able to restrict users from running long-running queries.

Reply:
Maybe in the setting then it can be something like maxTimeMSUpperLimit?

Reply:
In any way, I got this is wrong here and remove it!

Reply:
It could be `maxMaxTimeMS` :stuck_out_tongue_winking_eye: I mean, yeah, happy to rename it in #3768, I feel like `maxTimeMSUpperLimit` is pretty verbose since it already mentions that it’s a maximum in the name, but I do see your point